client/my-sites/backup/backup-contents-page/file-browser/file-browser-node.tsx

```diff
@@ -0,0 +1,94 @@
+import { Button, Icon } from '@wordpress/components';
+import { chevronDown, chevronRight } from '@wordpress/icons';
+import { FunctionComponent, useCallback, useState } from 'react';
+import FileTypeIcon from './file-type-icon';
+import { FileBrowserItem } from './types';
+import { useBackupContentsQuery } from './use-backup-contents-query';
+
+interface FileBrowserNodeProps {
+	item: FileBrowserItem;
+	path: string;
+	siteId: number;
+	rewindId: number;
+}
+
+const FileBrowserNode: FunctionComponent< FileBrowserNodeProps > = ( {
+	item,
+	path,
+	siteId,
+	rewindId,
+} ) => {
+	const isRoot = path === '/';
+	const isDirectory = item.type === 'dir';
```

Comment:
I wonder if the click controls should be keyed on hasChildren instead of the file type.  It won't change behavior if we use the transform to change the type, and things like plugin directories that return empty arrays when queried still show `has_children` true in the API.

Reply:
Just updated it to handle the click controls for now using `hasChildren` and not validating the file type. This way we could attempt to expand plugins and themes as well.

We might need to change this a bit when we introduce the logic to get view the file info.